### PR TITLE
update SDK dependency to 100.14 daily build

### PR DESCRIPTION
**Pull request by puneet-pdx**
- update dependencies version
- update artifact.gradle
- remove unnecessary inclusion of kotlin stdlib https://kotlinlang.org/docs/whatsnew14.html#dependency-on-the-standard-library-added-by-default 
- upgrade gradle plugin from 4.0.0 to 7.0.2 
     - upgrade the Android Gradle plugin from 4.0.0 to 7.0.2
     - upgrading the plugin forces us to upgrade our compiled SDK version from 29 to 31
     - ~upgrading the plugin forces us to upgrade Java Version from 1.8 to 11~
     - upgrading the plugin also wants the URL to be secured http -> https
        'http://olympus.esri.com/artifactory/arcgisruntime-repo/'

### CODE REVIEW

**Review by gunt0001 — COMMENTED** [review]
@puneet-pdx - looks good. Just the gradle wrapper scripts need updating.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -17,6 +17,6 @@
 #Tue Aug 27 09:42:42 BST 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (gunt0001) [inline gradle/wrapper/gradle-wrapper.properties:20]: You will need to update the `gradlew` and `gradlew.bat` files.

https://docs.gradle.org/current/userguide/gradle_wrapper.html

Comment (puneet-pdx) [reply gradle/wrapper/gradle-wrapper.properties]: updated

**Review by gunt0001 — APPROVED** [review]
@puneet-pdx - looks good.

**Pull request merged**